### PR TITLE
cut: slice flow-lint screen/flow/test graph story into a task plan

### PR DESCRIPTION
**Pull request by JMBattista**
## Summary
EPIC #404 (Screens & flows as first-class Smithy feature kinds) → Screens and Flows as UI Feature Kinds spec → User Story 6: flow-lint validates the screen/flow/test graph in app CI.
This `cut` decomposes US6 into a three-slice task plan — (1) a deterministic screen/flow/test graph resolver + validator with fixtures, (2) a `smithy flow-lint` CLI surface with CI-friendly exit codes and specific diagnostics, (3) source-side CI-adoption docs and helper-skill alignment — and links the plan from the spec's Dependency Order ledger. Cut is a decomposition step: implementation checkboxes stay unchecked (forge work), and the ledger Artifact column flipping from `—` to the new tasks.md path is this step's completion signal.

## Spec debt & uncertainty
- SD-005 (inherited): fidelity of `import`-mode structure derivation — how reliably render extracts screens/flows from a prototype and how much human confirmation is needed (owned by US5; flow-lint only validates the confirmed graph).
- SD-006 (inherited): whether SC/FL nodes are always atomic or can be sub-sliced (and whether `flow-scaffold` #410 is in scope) — shapes how cut slices UI nodes.
- SD-007 (inherited): build-phase coverage honesty — a build screen can be "done" with a missing brief state and no executable gate; flow-lint checks graph integrity, not per-state coverage.
- SD-008 (inherited): visual-intent honesty under the non-blocking gate (owned by US4); flow-lint does not validate visual-intent fulfillment.
- Assumption (from spec): `cut` slices every node kind into a tasks.md, and US6 is a backend-style CLI story, so it was sliced as a normal multi-slice story rather than a single SC/FL node.
- Uncertainty: slice boundaries (resolver → CLI → docs) are a judgment call; the resolver/validator split in Slice 1 is the load-bearing seam and could alternatively be two slices — kept as one for independent fixture-level reviewability.

## Validation
Planning-artifact-only change (Markdown); no code touched. Verified schema conformance: 4-column Dependency Order table (no checkboxes), slice structure (Goal/Justification/Addresses/acceptance criteria/PR Outcome), inherited Spec Debt rows, and spec-ledger US6 Artifact link. All four US6 acceptance scenarios (AS 6.1–6.4) and FR-025/FR-026 are mapped across the slices.

### CODE REVIEW

File: specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+**Justification**: The resolver is the load-bearing behavior behind the command. It is independently reviewable through fixture-level failures for dangling `screens:` references, missing test-body files, orphan test bodies, and duplicate flat IDs before CLI and CI ergonomics are added.
+
+**Addresses**: FR-025, FR-026; AS 6.1, AS 6.2, AS 6.3, AS 6.4
+
+### Tasks
+
+- [ ] **Discover durable UI artifacts from the repo tree**
+
+  Add a pure graph-discovery path that walks the selected repository root and reads `design/screens/*.design.md` and `design/flows/*.flow.md` artifacts using their YAML front matter. The discovery result should record each screen `id`, flow `id`, flow `screens` references, and flow `test-body` path, while treating the repo filesystem as the source of truth instead of any smithy session state.
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md:22]: Fixed in 520fa91: Slice 1's discovery task now defines the candidate test-body universe explicitly — the authoritative set is the `test-body:` paths declared across all `design/flows/*.flow.md` (the declared path is the contract per `smithy.helper-flow-definition`). Flow-lint must not blind-scan the project's test tree; the reverse orphan direction is evaluated only against an explicitly supplied flow-test root (or the conventional stub location), never the whole repo, and is reported as not-run when no root is available. Your `tests/e2e/Checkout.spec.ts` example is exactly the false-positive this rule prevents. The residual (fully general stray-test-body detection across arbitrary driver layouts) is tracked as SD-010.

File: specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  _Acceptance criteria:_
+  - Screen annotations are discovered from `design/screens/*.design.md`
+  - Flow definitions are discovered from `design/flows/*.flow.md`
+  - Flow discovery reads `id`, `screens`, and `test-body` fields
+  - Discovery can run from an arbitrary repo root or subpath
+  - Discovery performs no agent, network, or forge-specific calls
+
+- [ ] **Validate graph references and uniqueness**
+
+  Implement the `flow-lint` validation rules over the discovered graph. A `.flow.md` must fail when any `screens:` entry has no matching screen annotation, when its `test-body` file is missing, when an executable test body exists without a matching `.flow.md`, or when a flat `ScreenId` or `FlowId` is reused in the repo.
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md:37]: Fixed in 520fa91: added an explicit candidate-test-body discovery rule to Slice 1. The universe of flow test bodies is the declared `test-body:` set from `design/flows/*.flow.md`; the "and vice versa" (orphan) direction is scoped to a supplied flow-test root / conventional stub location rather than a blind driver-tree scan, so orphan detection is well-defined and false-positive-free. Residual openness recorded as SD-010.

File: specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+| S1 | Resolve the screen/flow/test artifact graph | — | — |
+| S2 | Expose `flow-lint` as a fast CLI check | S1 | — |
+| S3 | Document CI adoption and template boundaries | S2 | — |
+
+### Cross-Story Dependencies
+
+| Dependency | Direction | Notes |
+|------------|-----------|-------|
+| User Story 1: mark authors a UI spec with a typed ordering ledger + durable artifacts | depends on | Flow-lint validates the screen and flow artifact graph authored by mark. |
+| User Story 2: Tool-agnostic screen/flow generation from the project's own stack | depends on | Flow-lint consumes the driver-neutral `test-body` field and stable ID conventions established by the helper contracts. |
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md]: Fixed in 520fa91: reclassified the US2 cross-story row from "depends on" to "related". The `test-body` field and stable ID conventions come from the helper contracts — authoritative inputs from the closed sub-issues per the spec's Assumptions — not from US2's generation work, so US6 does not gate on US2. The row note now cites the spec ledger's authoritative `Depends On: US1` for US6, keeping the tasks file consistent with the parent ledger that planning/status tooling reads.

File: specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+---
+
+## Slice 1: Resolve the Screen/Flow/Test Artifact Graph
+
+**Goal**: Add a deterministic `flow-lint` graph resolver that can discover durable screen annotations, flow definitions, and paired executable test bodies, then validate the cross-reference rules without relying on smithy runtime state.
+
+**Justification**: The resolver is the load-bearing behavior behind the command. It is independently reviewable through fixture-level failures for dangling `screens:` references, missing test-body files, orphan test bodies, and duplicate flat IDs before CLI and CI ergonomics are added.
+
+**Addresses**: FR-025, FR-026; AS 6.1, AS 6.2, AS 6.3, AS 6.4
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md:16]: Not addressed: `AS 6.N` is the established cross-artifact convention in this repo — every sibling tasks file uses the same shorthand (`AS 5.1`, `AS 3.1`, …), and `AS 6.1`–`AS 6.4` map 1:1 and in order to User Story 6's four numbered Acceptance Scenarios in the spec. Rewriting only US6 to a different reference style would break that consistency without improving traceability. Leaving as-is intentionally.

File: specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+---
+
+## Slice 2: Expose `flow-lint` as a Fast CLI Check
+
+**Goal**: Wire the graph validator into a user-facing `smithy flow-lint` command that exits nonzero on findings, prints precise human-readable diagnostics, and exits cleanly for a fully resolved tree.
+
+**Justification**: The user story requires an invocable app-CI check, not only internal validation code. This slice turns the resolver into a command surface with stable exit behavior and clear diagnostics while keeping implementation separate from CI template examples.
+
+**Addresses**: FR-026; AS 6.1, AS 6.2, AS 6.3, AS 6.4
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md:67]: Not addressed: same as the line-16 thread — `AS 6.N` is the repo-wide tasks-file convention (matches `AS 5.N`/`AS 3.N` in the sibling files) and maps 1:1 to US6's numbered scenarios 1–4 in the spec. Kept for cross-artifact consistency.

File: specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+---
+
+## Slice 3: Document CI Adoption and Template Boundaries
+
+**Goal**: Document `flow-lint` as an app-repo CI check and align source-template guidance so durable screen/flow helper contracts point users to the command for cross-reference validation without making forge own linting state.
+
+**Justification**: The command only becomes useful when app maintainers know where it belongs. This slice keeps adoption guidance source-side, avoids regenerating deployed snapshots, and protects the architectural boundary that `flow-lint` is stateless and independent of forge.
+
+**Addresses**: FR-025, FR-026; AS 6.4
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply specs/2026-06-06-012-screens-and-flows-as-ui-feature-kinds/06-flow-lint-validates-the-screen-flow-test-graph-in-app-ci.tasks.md:114]: Not addressed: same as the other numbering threads — `AS 6.4` is the repo-wide shorthand for US6's fourth numbered Acceptance Scenario and matches the convention used across every sibling tasks file. Kept for consistency.
